src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -74,314 +38,540 @@ impl Color {
+/// based on environment variables.
+///
+/// # Arguments
+///
+/// * `err` - The clap Error to handle
+/// * `exit_code` - The exit code to use when exiting
+///
+/// # Panics
+///
+/// This function never returns - it always calls `std::process::exit()`.
+///
+/// # Examples
+///
+/// ```no_run
+/// use clap::Command;
+/// use uucore::clap_localization::handle_clap_error_with_exit_code;
+///
+/// let cmd = Command::new("myutil");
+/// match cmd.try_get_matches() {
+///     Ok(matches) => { /* handle matches */ },
+///     Err(e) => handle_clap_error_with_exit_code(e, 1),
+/// }
+/// ```
+pub fn handle_clap_error_with_exit_code(err: Error, exit_code: i32) -> ! {
+    let formatter = ErrorFormatter::new(crate::util_name());
+    formatter.print_error_and_exit(&err, exit_code);
 }
 
-impl LocalizedCommand for Command {
-    fn get_matches_localized(self) -> ArgMatches {
-        self.try_get_matches()
-            .unwrap_or_else(|err| handle_clap_error_with_exit_code(err, crate::util_name(), 1))
-    }
-
-    fn get_matches_from_localized<I, T>(self, itr: I) -> ArgMatches
-    where
-        I: IntoIterator<Item = T>,
-        T: Into<OsString> + Clone,
-    {
-        self.try_get_matches_from(itr)
-            .unwrap_or_else(|err| handle_clap_error_with_exit_code(err, crate::util_name(), 1))
-    }
+/// Configures a clap Command with proper localization and color settings.
+///
+/// This function sets up a Command with:
+/// - Appropriate color settings based on environment variables (NO_COLOR, CLICOLOR_FORCE, etc.)
+/// - Localized help template with proper formatting
+/// - TTY detection for automatic color enabling/disabling
+///
+/// # Arguments
+///
+/// * `cmd` - The clap Command to configure
+///
+/// # Returns
+///
+/// The configured Command with localization and color settings applied.
+///
+/// # Environment Variables
+///
+/// The following environment variables affect color output:
+/// - `NO_COLOR` - Disables all color output
+/// - `CLICOLOR_FORCE` or `FORCE_COLOR` - Forces color output even when not in a TTY
+/// - `TERM` - If set to "dumb", colors are disabled in auto mode
+///
+/// # Examples
+///
+/// ```no_run
+/// use clap::Command;
+/// use uucore::clap_localization::configure_localized_command;
+///
+/// let cmd = Command::new("myutil")
+///     .arg(clap::Arg::new("input").short('i'));
+/// let configured_cmd = configure_localized_command(cmd);
+/// ```
+pub fn configure_localized_command(mut cmd: Command) -> Command {
+    let color_choice = get_color_choice();
+    cmd = cmd.color(color_choice);
+
+    // For help output (stdout), we check stdout TTY status
+    let colors_enabled = match color_choice {
+        clap::ColorChoice::Always => true,
+        clap::ColorChoice::Never => false,
+        clap::ColorChoice::Auto => {
+            use std::io::IsTerminal;
+            IsTerminal::is_terminal(&std::io::stdout())
+                && std::env::var("TERM").unwrap_or_default() != "dumb"
+        }
+    };
```

Comment:
This snippet is almost identical with the `should_use_color_stderr` function, the only difference is in what is passed to `is_terminal`. Maybe that function can be made more flexible so that it can be used here, too?